### ISSUE TITLE
Oracle Linux on OKE cluster, <jemalloc>: Unsupported system page size

### ISSUE THREAD

**Issue by createchange** [fluent/fluent-bit#5778]
## Bug Report

First , want to reference #4270. 

I am using ARM64 Oracle Linux 7.9 on an Oracle Cloud Kubernetes cluster.

I tried installing the Sumo Logic's Helm chart (https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/main/deploy/helm/sumologic/Chart.yaml), but received the below error message.

Upon that failed install, I saw #4270, which clued me in that there was a fix. I looked at the Sumo Logic Helm chart to reference what release of fluent-bit was being used, and saw that it was an older release - 0.20.2 (though, from what I saw, still a release that should include the fixes from #4270). Upon that discovery, I attempted installing the latest release of the fluent-bit chart directly. I received the same error.

Brief snippet of error message:
```
<jemalloc>: Unsupported system page size
Error in GnuTLS initialization: ASN1 parser: Element was not found.
<jemalloc>: Unsupported system page size
<jemalloc>: Unsupported system page size
<jemalloc>: Unsupported system page size
malloc: Cannot allocate memory
```
Pod logs / error message is attached: [fluentbit-fluent-bit-5z9r4.log](https://github.com/fluent/fluent-bit/files/9174228/fluentbit-fluent-bit-5z9r4.log)

Node output:
```
[root@oke-cs56bxasraq-ntlw7x5fjfq-sbmqgfzjhbq-0 /]# getconf PAGESIZE
65536
```

Environment details:

* Version of fluent-bit used:
- Sumo Logic uses: 0.20.2
- I also tried directly installed fluent-bit Helm Chart version: 0.20.4
* Kubernetes version:  v1.23.4

* Node:
```
[root@oke-cs56bxasraq-ntlw7x5fjfq-sbmqgfzjhbq-0 /]# uname -a
Linux oke-cs56bxasraq-ntlw7x5fjfq-sbmqgfzjhbq-0 5.4.17-2136.307.3.1.el7uek.aarch64 #2 SMP Mon May 9 17:34:42 PDT 2022 aarch64 aarch64 aarch64 GNU/Linux
[root@oke-cs56bxasraq-ntlw7x5fjfq-sbmqgfzjhbq-0 /]# cat /etc/*elease
Oracle Linux Server release 7.9
NAME="Oracle Linux Server"
VERSION="7.9"
ID="ol"
ID_LIKE="fedora"
VARIANT="Server"
VARIANT_ID="server"
VERSION_ID="7.9"
PRETTY_NAME="Oracle Linux Server 7.9"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:oracle:linux:7:9:server"
HOME_URL="https://linux.oracle.com/"
BUG_REPORT_URL="https://bugzilla.oracle.com/"

ORACLE_BUGZILLA_PRODUCT="Oracle Linux 7"
ORACLE_BUGZILLA_PRODUCT_VERSION=7.9
ORACLE_SUPPORT_PRODUCT="Oracle Linux"
ORACLE_SUPPORT_PRODUCT_VERSION=7.9
Red Hat Enterprise Linux Server release 7.9 (Maipo)
Oracle Linux Server release 7.9
```

**To Reproduce**
- Have K8s cluster using Oracle Linux image for nodes
- Try to install Sumo Logic helm chart or latest release of fluent-bit directly

Thanks for any help you can provide!

**Comment by patrick-stephens** [comment]
There was a previous change to update the page size for ARM targets based on CentOS and Fedora defaults:
Can you confirm the Fluent Bit version actually being deployed to see if that change has gone in and/or use the latest?

If Oracle has decided to use a different page size then there's not much we can do - Oracle do provide two different kernels I believe: one to be RHEL compatible and one to be different although that may be for more recent versions.

**Comment by patrick-stephens** [comment]
Thinking about it, is it the container itself needs that option during compilation? That may be tricky to add as the containers are intended to be generic (having one coupled to a specific OS host and architecture). We could potentially add it to the UBI images we make separately for RHCC on ARM.

Probably need to confirm if compiling the container with the options from that PR above (#4270) resolves it?

**Comment by winston0410** [comment]
@patrick-stephens I have just tried the `1.9.9` image in Oracle Cloud's ARM instance, and I still have the same issue:

```
<jemalloc>: Unsupported system page size
```

I don't really know C and the fluentbit. What information can I provide to facilitate a debug here for you?

**Comment by winston0410** [comment]
@patrick-stephens Sorry it took me some time to set up bastion to connect with worker nodes correctly. This is the value I got:

```sh
[opc@oke]$ getconf PAGESIZE
65536
```

I am going to try and change the kernal next and see if it would work

**Comment by winston0410** [comment]
@patrick-stephens Seems like I am really out of lucky, RHCK is not avaliable for ARM instance https://blogs.oracle.com/scoter/post/oracle-linux-and-unbreakable-enterprise-kernel-uek-releases

**Comment by winston0410** [comment]
And seems like they have addressed this issue in UEK R7

https://docs.oracle.com/en/operating-systems/uek/7/relnotes7.0/uek7.0-NewFeaturesandChanges.html#uek-features-changes

**Comment by patrick-stephens** [comment]
OK, so the issue is page size and unfortunately this is 4k for the containers. It sounds like even Oracle have realised they need to resolve this but the only other thing I can suggest is building the containers with the options to modify the page size - I don't think we're going to support a myriad of page sizes for the official container images (we'd have to do tags for each one, etc.).

Potentially we could have page sizes as sub-registries in ghcr.io: ghcr.io/fluent/fluent-bit/64kpagesize:1.9.9.
It looks horrible though...

**Comment by tanmx** [comment]
@patrick-stephens I can't get the ghcr.io/fluent/fluent-bit/64kpagesize:1.9.9
```
# docker pull ghcr.io/fluent/fluent-bit/64kpagesize:1.9.9
Error response from daemon: Head "https://ghcr.io/v2/fluent/fluent-bit/64kpagesize/manifests/1.9.9": denied
```

**Comment by patrick-stephens** [comment]
Yeah I didn't say it had been created, just a possible naming scheme if we started building a myriad of matrix options for different things.

**Comment by JaredTan95** [comment]
I have the same issue, My Linux page size is `65535`:
<img width="437" alt="企业微信截图_3ef11479-41c9-48bc-b49c-dbb6ca49ed6d" src="https://user-images.githubusercontent.com/12468337/221138025-77c17c40-67ba-427c-9729-43a44bafc35d.png">

I upgrade fluent bit from `2.0.5` to `2.0.9`. But `2.0.5` works well. That's wired.

**Comment by patrick-stephens** [comment]
@JaredTan95 I don't believe anything has changed for the containers between 2.0.5 and 2.0.9 - are you sure you've pulled an ARM64 container? The platform is not part of the cache. Can you provide the full output of the logs and the metadata or the container? 

If you mean for packages then that's another deal entirely, but you don't provide any details on the actual target or the logs from  Fluent Bit.

Also, please don't provide screenshots - text is much easier to handle.

**Comment by JaredTan95** [comment]
@patrick-stephens hi, thx for your reply, I reproduced it again:

- docker metadata:
```bash
[root@host-x-x-x-x ~]# docker pull fluent/fluent-bit:2.0.9
2.0.9: Pulling from fluent/fluent-bit
5f80a38cb015: Pull complete
a839fcbfad1b: Pull complete
4d704a88b82e: Pull complete
e40cb4c669fc: Pull complete
0d2a2f1491cc: Pull complete
71b31c8271be: Pull complete
Digest: sha256:d370ad358bc8c392d157ed89eaf4555d49041bf86b29a6a8150981d04d286be8
Status: Downloaded newer image for fluent/fluent-bit:2.0.9
docker.io/fluent/fluent-bit:2.0.9
[root@host-x-x-x-x ~]# docker inspect docker.io/fluent/fluent-bit:2.0.9
[
    {
        "Id": "sha256:f35881e560a62e83b174d49dad6aa8e80deb547ea7938132358beae8714ab4f4",
        "RepoTags": [
            "fluent/fluent-bit:2.0.9"
        ],
        "RepoDigests": [
            "fluent/fluent-bit@sha256:d370ad358bc8c392d157ed89eaf4555d49041bf86b29a6a8150981d04d286be8"
        ],
        "Parent": "",
        "Comment": "buildkit.dockerfile.v0",
        "Created": "2023-02-03T16:04:31.653983161Z",
        "Container": "",
        "ContainerConfig": {
            "Hostname": "",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": null,
            "Cmd": null,
            "Image": "",
            "Volumes": null,
            "WorkingDir": "",
            "Entrypoint": null,
            "OnBuild": null,
            "Labels": null
        },
        "DockerVersion": "",
        "Author": "Bazel",
        "Config": {
            "Hostname": "",
            "Domainname": "",
            "User": "0",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "ExposedPorts": {
                "2020/tcp": {}
            },
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "SSL_CERT_FILE=/etc/ssl/certs/ca-certificates.crt",
                "FLUENT_BIT_VERSION=2.0.9"
            ],
            "Cmd": [
                "/fluent-bit/bin/fluent-bit",
                "-c",
                "/fluent-bit/etc/fluent-bit.conf"
            ],
            "ArgsEscaped": true,
            "Image": "",
            "Volumes": null,
            "WorkingDir": "/",
            "Entrypoint": [
                "/fluent-bit/bin/fluent-bit"
            ],
            "OnBuild": null,
            "Labels": {
                "author": "Eduardo Silva <eduardo@calyptia.com>",
                "description": "Fluent Bit multi-architecture container image",
                "org.opencontainers.image.authors": "Eduardo Silva <eduardo@calyptia.com>",
                "org.opencontainers.image.created": "2023-02-03T14:43:34.440Z",
                "org.opencontainers.image.description": "Fast and Lightweight Logs and Metrics processor for Linux, BSD, OSX and Windows",
                "org.opencontainers.image.documentation": "https://docs.fluentbit.io/",
                "org.opencontainers.image.licenses": "Apache-2.0",
                "org.opencontainers.image.revision": "4c0ca4fc5f28a34ca7d80e18b0d529bf9d07b1c2",
                "org.opencontainers.image.source": "https://github.com/fluent/fluent-bit",
                "org.opencontainers.image.title": "fluent-bit",
                "org.opencontainers.image.url": "https://github.com/fluent/fluent-bit",
                "org.opencontainers.image.vendor": "Fluent Organization",
                "org.opencontainers.image.version": "2.0.9",
                "vendor": "Fluent Organization",
                "version": "2.0.9"
            }
        },
        "Architecture": "arm64",
        "Os": "linux",
        "Size": 80451506,
        "VirtualSize": 80451506,
        "GraphDriver": {
            "Data": {
                "LowerDir": "/var/lib/containers/docker/overlay2/08ee6c6efa1c01c6ce3487cc74134f9bad2a2e86ff42d47c9e3f106828288001/diff:/var/lib/containers/docker/overlay2/201e87556e76bfdffd04c4e2500fab0f319e17020d7f2a8585b67d94cf57c2f8/diff:/var/lib/containers/docker/overlay2/5bf8aba71a900b2275ad4fbb5eee3720a24e3ea0127f8f7c6600728822fc9b32/diff:/var/lib/containers/docker/overlay2/644c0b6c30b1416c41fa9b2baf383d581979917328f1f579a7a752fe66802771/diff:/var/lib/containers/docker/overlay2/fbc9b2076e8ba869ef5b3bab2aca64caa762e4d58ab5a0bb56bf9397d4c15105/diff",
                "MergedDir": "/var/lib/containers/docker/overlay2/a64e2612d3b0dbe6bc07c4631e89c341ec8559cfb463d33e1ef07e9bebd926c4/merged",
                "UpperDir": "/var/lib/containers/docker/overlay2/a64e2612d3b0dbe6bc07c4631e89c341ec8559cfb463d33e1ef07e9bebd926c4/diff",
                "WorkDir": "/var/lib/containers/docker/overlay2/a64e2612d3b0dbe6bc07c4631e89c341ec8559cfb463d33e1ef07e9bebd926c4/work"
            },
            "Name": "overlay2"
        },
        "RootFS": {
            "Type": "layers",
            "Layers": [
                "sha256:fba4381f2bb70de24aadfb483fb2b5254e81b0f8db1959526a1beec8370363bc",
                "sha256:96e320b34b5478d8b369ca43ffaa88ff6dd9499ec72b792ca21b1e8b0c55670f",
                "sha256:7e2cf97ed8c49989106a276ec387e30bbd564b3c1a8ff943a13fd0f3cb4ad4f9",
                "sha256:09163696f0f9de226b0d3f27fa78ae14fff8ae1d4131e7804f5f47859e133f8c",
                "sha256:0a8123fc64c303f94a3279ec74c6ace8eb7cbfec86620d888b5b53e2ada6cad0",
                "sha256:931d784870af82afb151c63d54d972b545a825fe95ba39365661f711226a2993"
            ]
        },
        "Metadata": {
            "LastTagTime": "0001-01-01T00:00:00Z"
        }
    }
]
```

- run an container:
```bash
[root@ ~]# docker run -ti docker.io/fluent/fluent-bit:2.0.9
<jemalloc>: Unsupported system page size
<jemalloc>: Unsupported system page size
.......
<jemalloc>: Unsupported system page size
Error in GnuTLS initialization: ASN1 parser: Element was not found.
<jemalloc>: Unsupported system page size
<jemalloc>: Unsupported system page size
<jemalloc>: Unsupported system page size
malloc: Cannot allocate memory
```

- linux pagesize:
```bash
[root@host-192-168-10-107 ~]# getconf PAGESIZE
65536
```

- linux arch
```bash
[root@host-192-168-10-107 ~]# uname -a
Linux host-192-168-10-107 4.19.90.S001.64.200410 #1 SMP Fri Aug 6 15:52:06 CST 2021 aarch64 aarch64 aarch64 GNU/Linux
```

**Comment by patrick-stephens** [comment]
What platform are you running on?

**Comment by patrick-stephens** [comment]
Right - so see https://github.com/fluent/fluent-bit/issues/5778#issuecomment-1265244244.
I would expect your platform to not work, that's what should happen as the page sizes do not match. I'm not sure why 2.0.5 is working for you though...

**Comment by JaredTan95** [comment]
> Right - so see [#5778 (comment)](https://github.com/fluent/fluent-bit/issues/5778#issuecomment-1265244244). I would expect your platform to not work, that's what should happen as the page sizes do not match. I'm not sure why 2.0.5 is working for you though...

I checked my platform page sizes:
```bash
[root@host-192-168-10-107 ~]# getconf PAGESIZE
65536
````

this is the root-cause. I think, But, the only way to solve this is that we need to build from source by myself?

**Comment by patrick-stephens** [comment]
Yes, unless you can think of a better option. 

The issue is the mismatch in page size which requires recompiling for the specific settings. We'd have to maintain multiple variants for every combination of architecture and page size (plus anything else the distribution deviates in) which i don't think is feasible to maintain or easily consume.

**Comment by jynolen** [comment]
Hello

I'm facing same issue here.
Don't know if it's viable or not. Main issue is that Jemalloc have to be built with specific options.
PR Ref #5027

Maybe this flag will make all arm64 build comptatible whatever the pagesize (What can do most can do less).
In the Dockerfiles buildx process maybe we could have a check to set this flag only when arch == arm64.

JY.

**Comment by patrick-stephens** [comment]
The issue is not that @jynolen , the issue is we have to pass a value to specify and unfortunately this value is different for Oracle Linux on ARM64 than it is for CentOS/RHEL on ARM 64. At the moment we only support CentOS so that's the only package there is.

It requires an entirely new Oracle-specific build to resolve with new packages and a new repository for them - and its only required for ARM64 as AMD64 uses the same page size.

**Comment by galovics** [comment]
just leaving this for future reference. I encountered this as well on ARM64 Oracle Linux 7.9. The new [UEK 7](https://docs.oracle.com/en/operating-systems/uek/7/relnotes7.0/uek7-AboutUEKR7.html) comes with a regular 4k pagesize for Oracle but it's only available for Oracle Linux 8 and 9.

A simple linux upgrade on our Kubernetes cluster would solve this issue supposedly but we encountered stability issues with Oracle Linux 8+ (and 9 is not yet available for OKE) so upgrading wasn't an issue for us. 

We compiled a customer version of fluent-bit that comes with 64k pagesize on arm64. If anyone needs it, it's here: https://github.com/docktape/fluent-bit/pkgs/container/fluent-bit/422657388?tag=v4.0.2-oke-7.9-arm64

It was based off-of the v4.0.2 tag.
